### PR TITLE
[PLGN-704] - rapid7_intsights - Fixing bad response parsing from the takedown request api

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - Updated the following
   - Made sure a valid 200  response no content response from takedown_request action did not throw  
  - Ran Black formator 
  - ran insight-pugin refresh 
  - updated unit tests 
    - added new tests for takedown request 
    - fixed old imports
    - added schema validation to all tests

https://rapid7.atlassian.net/browse/PLGN-704

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

New unit tests have been added for the takedown request action 

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

testing a valid request now will return correctly and not error
image

if the alert has already had the remediation started an more useful error message is now returned 
image

if the alert cannot be remediated a more useful error message is now returned 
image

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/rapid7_intsights/help.md [hunk 1]
```diff
@@ -833,142 +445,206 @@ Example output:
...
-      "ConfidenceLevel": 2,
-      "Name": "Cyber Threat Alliance"
-    }
-  ],
-  "system_tags": [],
-  "tags": [],
-  "type": "Domains",
-  "value": "example.com",
-  "whitelist": true
+  "content": "{ \"content\": [ { \"cveId\": \"CVE-2020-7064\", \"cpe\": [ { \"Value\": \"cpe:2.3:a:php:php:*:*:*:*:*:*:*:*\", \"Range\": { \"VersionStartIncluding\": \"7.2.0\", \"VersionEndIncluding\": \"\", \"VersionStartExcluding\": \"\", \"VersionEndExcluding\": \"7.2.9\" }, \"Title\": \"Php\", \"VendorProduct\": \"Php Php\" } ], \"publishedDate\": \"2020-04-01T04:15:00.000Z\", \"updateDate\": \"2021-10-25T10:14:52.978Z\", \"severity\": \"Low\", \"intsightsScore\": 36, \"cvssScore\": 5.4, \"mentionsAmount\": 39, \"mentionsPerSource\": { \"PasteSite\": 0, \"HackingForum\": 0, \"InstantMessage\": 0, \"DarkWeb\": 0, \"ClearWebCyberBlogs\": 0, \"CodeRepositories\": 9, \"Exploit\": 0, \"SocialMedia\": 30 }, \"firstMentionDate\": \"2020-03-19T15:09:00.000Z\", \"lastMentionDate\": \"2021-07-22T20:41:00.000Z\", \"exploitAvailability\": false, \"vulnerabilityOrigin\": [ \"API\" ], \"relatedThreatActors\": [], \"relatedMalware\": [], \"relatedCampaigns\": [] } ], }"
```
Comment (igorski-r7) [inline plugins/rapid7_intsights/help.md:634]: Could you please pretty print it?

File: plugins/rapid7_intsights/unit_test/test_get_cve_by_id.py [hunk 2]
```diff
@@ -18,31 +19,44 @@ def setUpClass(cls, mock_request) -> None:
...
         expected = Util.read_file_to_dict("expecteds/get_cve_by_id_all.json.resp")
         self.assertEqual(expected, actual)
+        validate(actual, GetCveByIdOutput.schema)
 
     @patch("requests.request", side_effect=Util.mock_request)
     def test_get_cve_with_offset(self, make_request):
-        self.maxDiff = None
-        actual = self.action.run({Input.CVE_ID: ["CVE-2021-7064"]})
+        input_params = {Input.CVE_ID: ["CVE-2021-7064"]}
+        validate({}, GetCveByIdInput.schema)
```
Comment (joneill-r7) [inline plugins/rapid7_intsights/unit_test/test_get_cve_by_id.py:31]: copy and paste issue? validating `{}` and not input_params? On this test and the next couple

File: plugins/rapid7_intsights/help.md (hunk 1 above)
Comment (rbowden-r7) [reply plugins/rapid7_intsights/help.md:634]: This was due to the example in the plugin spec being a string and not a dict, this has been updated

File: plugins/rapid7_intsights/help.md [hunk 3]
```diff
@@ -1317,211 +1178,216 @@ Example output:
...
-|Name|string|False|Name of the technique|
-|URL|string|False|URL|
 
 ## Troubleshooting
 
 _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 5.1.1 - Better handling of response fromt he threat connect API when using the `takedown_request` action | Bumped to use the newest version of the SDK | Updated old unit tests / added new unit tests
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/rapid7_intsights/help.md:1390]: This has been updated

File: plugins/rapid7_intsights/unit_test/test_get_cve_by_id.py (hunk 2 above)
Comment (rbowden-r7) [reply plugins/rapid7_intsights/unit_test/test_get_cve_by_id.py]: yeah good catch, this copy paste error has been addressed

File: plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py [hunk 4]
```diff
@@ -283,30 +286,47 @@ def make_request(self, method: str, path: str, json_data: dict = None, params: d
...
+            self.check_status_codes(response=response)
 
             if 200 <= response.status_code < 300:
                 return response
 
             raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
         except requests.exceptions.HTTPError as e:
             raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)
+
+    def check_status_codes(self, response: requests.Response):
```
[the remark this answers is not in the document]
Comment (rbowden-r7) [reply plugins/rapid7_intsights/icon_rapid7_intsights/util/api.py]: This has been updated to remove self and added as a static method

**Review by joneill-r7 — APPROVED** [review]

**Review by ablakley-r7 — APPROVED** [review]

**Pull request merged**